src/client/renderer/r_bloom.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright(c) 1997-2001 id Software, Inc.
+ * Copyright(c) 2002 The Quakeforge Project.
+ * Copyright(c) 2006-2011 Quetoo.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+
+#include "r_local.h"
+
+/**
+ * @brief The bloom vertex type.
+ */
+typedef struct {
+	vec2s_t position;
+	vec2_t texcoord;
+} r_bloom_vertex_t;
+
+/**
+ * @brief The bloom data.
+ */
+static struct {
+	GLuint vertex_array;
+	GLuint vertex_buffer;
+} r_bloom_data;
+
+/**
+ * @brief The bloom program.
+ */
+static struct {
+	GLuint name;
+
+	GLuint uniforms_block;
+
+	GLint in_position;
+	GLint in_texcoord;
+
+	GLint texture_color_attachment;
+	GLint texture_bloom_attachment;
+} r_bloom_program;
+
+/**
+ * @brief
+ */
+void R_DrawBloom(const r_view_t *view) {
```

Comment:
So this is pretty straightforward:

1. Keep rendering to the framebuffer's primary color attachment
2. Bind both the color attachment and bloom attachment to textures
3. Draw a full-screen quad, that re-writes the color attachment and blurs and blends the bloom attachment over it.